src/status_im2/contexts/chat/composer/effects.cljs

```diff
@@ -105,9 +105,23 @@
   (.remove ^js @keyboard-hide-listener)
   (.remove ^js @keyboard-frame-listener))
 
+(defn max-height-effect
+  [{:keys [focused?]}
+   {:keys [max-height]}
+   {:keys [height saved-height last-height]}]
+  (rn/use-effect
+   (fn []
+     (when @focused?
+       (let [new-last-height (min max-height (reanimated/get-shared-value last-height))]
```

Comment:
can be renamed to `new-height`. 

Also, are you sure this statement is needed? Setting the height and saved-height to value of last-height in the effect instead of on-focus won't be sufficient?

Reply:
> can be renamed to new-height.

Will do it :)

> Also, are you sure this statement is needed? Setting the height and saved-height to value of last-height in the effect instead of on-focus isn't sufficient?

Indeed @OmarBasem. The critical difference is that this effect runs when `max-height` OR `focused?` are changed, whereas `on-focus` runs only when it needs to focus and that means that it would not be using the most up-to-date max-height value because the `max-height` is calculated *slightly* after `on-focus` runs (right after subscriptions are changed).

It's easy to see:

1. Paste a large text and one unfurleable URL or an add an image.
2. Go out of the chat.
3. Open the chat again. The `on-focus` event will be triggered before the link preview or image subscription arrives. That means `on-focus` will use the outdated `max-height` and whatever happens inside `on-focus` won't react to the new `max-height` value, and that's why the UI will look broken.

Reply:
Renamed and added a quick comment in the code. https://github.com/status-im/status-mobile/pull/15989/commits/3e8b7620bb92fe91704bc528d028697f53cc6331